src/datachain/catalog/catalog.py

```diff
@@ -1198,40 +1236,77 @@ def get_remote_dataset(
 
     def get_dataset_dependencies(
         self,
-        name: str,
-        version: str,
+        name: str | None = None,
+        version: str | None = None,
         namespace_name: str | None = None,
         project_name: str | None = None,
         indirect=False,
+        dataset_id: int | None = None,
```

Comment:
why do we need this API change?

Reply:
If we already have dataset id with us, we avoid an extra API for Studio (API metastore), one extra database call to get the dataset. 
This is helpful for optimizing in studio.